source/slang/slang-check-conformance.cpp

```diff
@@ -51,19 +51,23 @@ namespace Slang
 
     SubtypeWitness* SemanticsVisitor::isSubtype(
         Type* subType,
-        Type* superType)
+        Type* superType,
+        bool subtypeInheritanceIsNotFullyResolved
```

Comment:
I would like to understand why this additional parameter is needed. When this parameter is added, how does any incoming code know when to set it to true and when to set it to false?

Instead of introducing that question to every callsite, we should try to avoid requiring it.

It seems that the intent here is for some situations we don't want the cache the result. But in that case it seems the right approach is to invalidate the cache entry when an inheritance decl is checked. Will there be a bigger problem when that inheritance decl is resolved after this function is called?

Reply:
what the param does:
1. no caching --> if we did not 'finish' the lookup stage and still are adding structs, we must not cache, we will have incorrect cache.
2. We do not 'ensure' the subtype decl, we assume it is safe to lookup. This is required for when we call the function at a time during/before the lookup stage. If we do not stop the ensureDecl, we will have a infinite recursive call.

Reply:
I like to suggest to refactor the new boolean variable.
It appears that the behavior around the new boolean variable is that when it is true, "isSubtype()" behaves same as before.
Only when the boolean variable is false, "isSubtype()" behaves little differently; I am not clear on how different tho.

If my observation is correct, you could make a new function like following without changing the existing function, "isSubtype()", 
```
SubtypeWitness * XXX:isSubtypeWhenInheritanceIsNotFullyResolved(Type* subType, Type* superType)
{
    auto result = isSubtype(subType, superType);
    if (result == null_ptr)
    {
       do something here...
    }
    return result;
}
```
The reason why I suggest this is that "boolean" parameter is not a good thing to have in general.
It can be only true or false, which can be same as having two distinct functions instead of having a single function with a boolean parameter.
For more information, I like to suggest a very good book called, "Clean code", written by Martin C Robert.
His another book, "Agile Software Development, Principles, Patterns, and Practices", is also a fantastic book to checkout.